ext/AvizExt/viz/clustering.jl

```diff
@@ -29,24 +32,162 @@ function ADRIA.viz.ts_cluster!(g::Union{GridLayout,GridPosition}, data::Abstract
+    data_stats = collect(dropdims(opts[:summary](data, dims=:timesteps), dims=:timesteps))
+
+    cluster_colors = _clusters_colors(clusters)
+    legend_params = _cluster_legend_params(clusters, cluster_colors, data_stats)
+
+    opts[:highlight] = get(opts, :highlight, cluster_colors)
+    opts[:legend_params] = get(opts, :legend_params, legend_params)
+
+    ADRIA.viz.map!(g, rs, data_stats; opts=opts, axis_opts=axis_opts)
 
     return g
 end
 
+"""
+    _cluster_colors(clusters::Vector{Int64})
+
+Vector of cluster colors.
+
+# Arguments
+- `clusters` : Vector of numbers corresponding to clusters
 
+# Returns
+Vector{RGBA{Float32}}
+"""
+function _clusters_colors(clusters::Vector{Int64})::Vector{RGBA{Float32}}
+    # Number of non-zero clusters
+    n_clusters = length(unique(filter(cluster -> cluster != 0, clusters)))
+
+    # Vector of clusters colors for non-zero clusters
+    clusters_colors = categorical_colors(:seaborn_bright, n_clusters)
+
+    # Assign color "black" to cluster == 0
+    rgba_black = parse(RGBA{Float32}, "transparent")
+    return [cluster == 0 ? rgba_black : clusters_colors[cluster] for cluster in clusters]
+end
+
+"""
+    _cluster_color(unique_cluster_colors::Vector{RGBA{FLoat32}}, clusters::Vector{Int64}, cluster::Int64)
+
+Color parameter for current cluster weighted by number of scenarios
+
+# Arguments
+- `unique_cluster_colors` : Vector of all cluster options that are being used
+- `clusters` : Vector of numbers corresponding to clusters
+- `cluster` : Current cluster
+
+# Returns
+Tuple{RGBA{Float32}, Float64}
+"""
+function _cluster_color(unique_cluster_colors::Vector{RGBA{Float32}},
+    clusters::Vector{Int64}, cluster::Int64)::Tuple{RGBA{Float32},Float64}
+    # Number of scenarios for a cluster
+    n_scens = count(clusters .== cluster)
+
+    # Compute line weight
+    color_weight = max(min((1.0 / (n_scens * 0.05)), 0.6), 0.1)
+
+    return (unique_cluster_colors[cluster], color_weight)
+end
+
+"""
+    _cluster_legend_params(clusters::Vector{Int64}, clusters_colors::Vector{RGBA{Float32}}, data_statistics::Vector{Float32})
+
+Color parameter for current cluster weighted by number of scenarios
+
+# Arguments
+- `clusters` : Vector of numbers corresponding to clusters
+- `cluster_colors` : Vector of all cluster options that are being used
+- `data_statistics` : Vector of statistics for each site (default is mean)
+
+# Returns
+Tuple{RGBA{Float32}, Float64}
+"""
+function _cluster_legend_params(clusters::Vector{Int64}, 
+    clusters_colors::Vector{RGBA{Float32}}, data_statistics::Vector{Float32})::Tuple
+    # Filter non-zero clusters from clusters, colors and data
+    non_zero_clusters = clusters .!= 0
+    clusters_filtered = clusters[non_zero_clusters]
+    colors_filtered = clusters_colors[non_zero_clusters]
+    statistics_filtered = data_statistics[non_zero_clusters]
+
+    # Fill legend entries colors
+    legend_entries = [PolyElement(color = color, strokecolor = :transparent) for 

```

Comment:
I'm surprised the formatter did not remove the whitespace in the argument keyword assignment (`color=color`)! Could you please re-format?

Reply:
@ConnectedSystems  My vscode was configured to use a formatter that was doing that. I have changed to the other Julia formatter I can choose here and, apparently, it is right now. Please let me know if you see something else that is wrong.

Reply:
UPDATE: apparently when I use this other format option it adds a lot of blank spaces at the end of some lines (I think we had the same problem in the past, with some spaces that we couldn't see at vscode but were appearing at GitKraken...). I am not sure what's going on, do you remember what we did when that happened a few weeks ago?

Reply:
I think just `trim` (ctrl+shift+p -> "trim" -> enter), I'll have to see what's going on next time we chat.

Reply:
I have already tried `trim`, it didn't work... I'll have to investigate further...